lib/pavilion/plugins/commands/show.py

```diff
@@ -385,65 +412,117 @@ def _functions_cmd(self, _, args):
     def _hosts_cmd(self, pav_cfg, args):
         """List all known host files."""
 
-        hosts = []
-        col_names = ['Name']
-        if args.verbose:
-            col_names.append('Path')
-        for conf_dir in pav_cfg.config_dirs:
-            path = conf_dir / 'hosts'
+        if args.config is None:
+            hosts = []
+            col_names = ['Name']
+            if args.verbose:
+                col_names.append('Path')
+            if args.vars:
+                col_names.append('Variables')
+            for conf_dir in pav_cfg.config_dirs:
+                path = conf_dir / 'hosts'
 
-            if not (path.exists() and path.is_dir()):
-                continue
+                if not (path.exists() and path.is_dir()):
+                    continue
 
-            for file in os.listdir(path.as_posix()):
+                for file in os.listdir(path.as_posix()):
+
+                    file = path / file
+                    if file.suffix == '.yaml' and file.is_file():
+                        host_id = file.stem
+                        host_path = file
+                        with file.open() as config_file:
+                            config_data = yc_yaml.load(config_file)
+                        try:
+                            host_vars = list(config_data['variables']
+                                             .keys())
+                        except (KeyError, TypeError) as err:
+                            host_vars = []
```

Comment:
You should load these with the test config loader, not as raw yaml.

```python
from pavilion.test_config import file_format

config = file_format.TestConfigLoader().load(myfile)
```

This will guarantee that all the config fields exist in the resulting structure, though they may be empty. So the 'variables' key will always exist, for instance, and will always be a dictionary.

Reply:
Ok, so the --vars output still shows what variables are available. 
  - But is a list of their names really readable? 
  - What are their defined sub-keys (if any)
  - What are their values? 
  - How do they compare across host files?

There's lots of stuff that would be useful to do here, and it's going to require a completely separate table from the standard one.